src/resolve/resolved-mdns.c

```diff
@@ -401,6 +398,15 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
         if (dns_packet_validate_reply(p) > 0) {
                 DnsResourceRecord *rr;
 
+                /* RFC 6762 section 6:
+                 * The source UDP port in all Multicast DNS responses MUST be 5353 (the well-known port
+                 * assigned to mDNS). Multicast DNS implementations MUST silently ignore any Multicast DNS
+                 * responses they receive where the source UDP port is not 5353. */
+                if (p->sender_port != MDNS_PORT) {
+                        log_trace("Received mDNS reply packet from port %u, ignoring.", p->sender_port);
```

Comment:
Personally I think it would be better if it was `log_debug` and was logged after `Got mDNS reply packet` to make it easier to spot fishy attempts to send those packets without recompiling `systemd`. Looking at
```sh
Aug 31 14:36:51 H systemd-resolved[10490]: Received mdns UDP packet of size 795, ifindex=3, ttl=255, fragsize=0, sender=192.168.56.3, destination=224.0.0.251
Aug 31 14:36:51 H systemd-resolved[10490]: Received mDNS reply packet from port 1234, ignoring.
```
I'm not sure why ports aren't logged in the first log message but it isn't that important anyway.

Reply:
Thank you for the suggestions. Applied, except for the ordering with `Got mDNS reply packet`. The newly added log message explicitly mentions that we received mDNS reply. It is not necessary to log duplicated information.

Reply:
>  It is not necessary to log duplicated information

Makes sense. Those events can be combined into one log message of course.